src/shared/calculations.js

```diff
@@ -9,46 +10,132 @@ import quantile from "@stdlib/stats/base/dists/beta/quantile";
-     * unknown) probability of success.
+     * Compute the proportion of reviews which gave a product a "positive" rating,
+     * either using the "avg" method (where the input is the average star rating) 
+     * or "dist" method (where the input is the 5-element discrete distribution
+     * from the Amazon rating percentage histogram).
+     *
+     * @param {Object.<string, (number|number[])>} ratingData - object containing
+     *     the average rating under key "avg" and/or the 5-element rating
+     *     distribution (fraction of ratings at each star value) under key "dist".
+     * @param {string} method - which method to use to compute proportion of
+     *     reviews which were positive; must be either "avg" or "dist".
+     * @returns {number} proportion - proportion of ratings which were positive.
+     */
+    getProportionPositive(ratingData, method = "avg") {
+        if (method == "avg") {
+            return this.getProportionPositiveFromAvg(ratingData["avg"]);
+        } else if (method == "dist") {
+            return this.getProportionPositiveFromDist(ratingData["dist"]);
+        } else {
+            throw `'method' must be either 'dist' or 'avg', not '${method}'`;
+        }
+    },
 
-     * @param {number} avgRating - average value of all star ratings.
-     * @param {number} numRatings - total number of ratings given.
+    /**
+     * Recast Amazon product ratings as a binomial distribution by converting the
+     * star ratings into positive/negative (binary) ratings. Assume that each
+     * converted binary rating is the result of an independent Bernoulli trial with
+     * fixed (but unknown) probability of success. By framing the problem this way,
+     * we can use a beta distribution (the conjugate prior distribution of the
+     * Bernoulli distribution) to quantify our knowledge of the underlying probability
+     * of success (i.e. the probability of having a "positive" experience with the
+     * Amazon product).
      *
+     * In this function, we use the rating data to compute the "alpha" and "beta"
+     * parameters for the beta distribution described above.
+     *
+     * @param {Object.<string, (number|number[])>} ratingData - object containing
+     *     the average rating under key "avg" and/or the 5-element rating
+     *     distribution (fraction of ratings at each star value) under key "dist".
+     * @param {number} numRatings - total number of ratings for this product.
+     * @param {string} method - which method to use to compute proportion of
+     *     reviews which were positive; must be either "avg" or "dist".
+     * @returns {Object.<string, number>} betaParams - object containing "alpha"
+     *     and "beta" parameters for the beta distribution, as well as the
+     *     estimated "proportion" and number "numPositive" of ratings which were 
+     *     positve, and the total number of ratings "numRatings".
+     */
+    getBetaParams(ratingData, numRatings, method = "avg") {
```

Comment:
(I just wrote a beautifully composed comment and lost it when I clicked a link. :cry: Here is but a memory of what was...)

At this point, we're accepting a parameter that is only used further down the call stack (only to determine which method to drive). There's also a bit of redundancy as the user has to specify "avg" or "dist" twice. I can't help but think there's a better way to get where we're going.

Perhaps we should just have two separate methods, like`getBetaParamsDist(...)` and `getBetaParamsDist(...)`?

The [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) would suggest that it's too early to try to DRY this anyway.

Reply:
If we really feel like it's necessary to have different "modes" then perhaps we should add a `mode` property.

Reply:
I get what you're saying; it does feel redundant, but I'm not exactly sure how to fix it. The idea was that `ratingData` would be a dictionary of all the possible forms of rating data you care about, like:
```
ratingData = {
  "dist": [0.1, 0, 0, 0.4, 0.5],
  "avg": 4.7
}
```
So you can just specify that once and decide later whether you want to use "avg" or "dist" to generate your statistics. Now that I say that though, if that's what I'm going for, it seems like I might also want to include `numRatings` inside of that dictionary...

Not sure where to go from here. Thoughts?

Reply:
I'm a huge believer in taking the dumb approach first and iterating from there as real-world problems arise. Would you be opposed to having an "average" and a "distribution" version of each of those functions, for now?